sdv/single_table/base.py

```diff
@@ -75,3 +95,638 @@ def fit(self, data):
+            column_values = sampled[column]
+            if column_values.dtype.kind == 'f':
+                distance = value * float_rtol
+                sampled = sampled[np.abs(column_values - value) <= distance]
+                sampled[column] = value
+            else:
+                sampled = sampled[column_values == value]
+
+        return sampled
+
+    def _sample_rows(self, num_rows, conditions=None, transformed_conditions=None,
+                     float_rtol=0.1, previous_rows=None):
+        """Sample rows with the given conditions.
+
+        Input conditions is taken both in the raw input format, which will be used
+        for filtering during the reject-sampling loop, and already transformed
+        to the model format, which will be passed down to the model if it supports
+        conditional sampling natively.
+
+        If condition columns are float values, consider a match anything that
+        is closer than the given ``float_rtol`` and then make the value exact.
+
+        If the model does not have any data columns, the result of this call
+        is a dataframe of the requested length with no columns in it.
+
+        Args:
+            num_rows (int):
+                Number of rows to sample.
+            conditions (dict):
+                The dictionary of conditioning values in the original format.
+            transformed_conditions (dict):
+                The dictionary of conditioning values transformed to the model format.
+            float_rtol (float):
+                Maximum tolerance when considering a float match.
+            previous_rows (pandas.DataFrame):
+                Valid rows sampled in the previous iterations.
+
+        Returns:
+            tuple:
+                * pandas.DataFrame:
+                    Rows from the sampled data that match the conditions.
+                * int:
+                    Number of rows that are considered valid.
+        """
+        if self._data_processor._dtypes is not None:
+            if conditions is None:
+                sampled = self._sample(num_rows)
+            else:
+                try:
+                    sampled = self._sample(num_rows, transformed_conditions)
+                except TypeError:
+                    sampled = self._sample(num_rows)
+
+            sampled = self._data_processor.reverse_transform(sampled)
+
+            if previous_rows is not None:
+                sampled = pd.concat([previous_rows, sampled], ignore_index=True)
+
+            sampled = self._data_processor.filter_valid(sampled)
+
+            if conditions is not None:
+                sampled = self._filter_conditions(sampled, conditions, float_rtol)
+
+            num_valid = len(sampled)
+
+            return sampled, num_valid
+
+        else:
+            sampled = pd.DataFrame(index=range(num_rows))
+            sampled = self._data_processor.reverse_transform(sampled)
+            return sampled, num_rows
```

Comment:
Do we know what this situation is for? Is this for anonymization?

Reply:
This is one of the open questions I had. I don't see when `data_processor` will not have `Dtypes` 🤔  (When not fitted?)

Reply:
Try running through the integration tests for `sdv/tabular/base.py` and see if any reach this point

Reply:
This case is when there are no other columns but `primary_keys` or `ids` . We will have to make some changes on the `data_processor` to get this case allowed.